### PR TITLE
136 create spotify module

### LINKED ISSUE #136: create spotify module
- [ ] create own spotify sdk
- [ ] implement share profile toggling
- [ ] handle `/profile/me` route

### PR DESCRIPTION
**Pull request by Mnigos**
closes #136

### CODE REVIEW

File: src/modules/spotify/auth/spotify-auth.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+      expect(await spotifyAuthService.token({ refreshToken })).toEqual(
+        spotifyTokenMock
+      )
+      expect(getSpy).toHaveBeenCalledTimes(4)
+      expect(postSpy).toHaveBeenCalled()
+    })
+  })
+
+  test('should get me profile', async () => {
```
Comment (Iqro-dev) [inline src/modules/spotify/auth/spotify-auth.service.spec.ts:81]:
```suggestion
  test('should get profile', async () => {
```

File: src/modules/users/users-playback.controller.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  ): Promise<Success> {
+    const foundUser = await this.usersRepository.findOneBy({ id })
+
+    if (!foundUser) throw new NotFoundException(NOT_BEEN_FOUND(USER))
+
+    const meProfile = await this.spotifyAuthService.getMeProfile(accessToken)
+
+    if (foundUser.profile.id !== meProfile.id)
+      throw new UnauthorizedException(
+        'You are not authorized to resume playback.'
```
Comment (Iqro-dev) [inline src/modules/users/users-playback.controller.ts:132]:
```suggestion
        'You are not allowed to resume playback.'
```

File: src/modules/users/users-profile.controller.ts [hunk 3]
```diff
@@ -230,18 +229,47 @@ export class UsersProfileController {
...
+  //   const foundUser = await this.usersRepository.findOneBy({ id })
+
+  //   if (!foundUser) throw new NotFoundException(NOT_BEEN_FOUND(USER))
+
+  //   const token = await this.spotifyAuthService.token({
+  //     refreshToken: foundUser.refreshToken,
+  //   })
+
+  //   return this.spotifyPlayerService.getPlaybackState(token)
+  // }
```
Comment (Iqro-dev) [inline src/modules/users/users-profile.controller.ts:274]: What's that?

**Review by Iqro-dev — CHANGES_REQUESTED** [review]

File: src/modules/spotify/auth/spotify-auth.service.spec.ts (hunk 1 above)
Comment (Mnigos) [reply src/modules/spotify/auth/spotify-auth.service.spec.ts:81]: this is done purposely to highlight that this is profile for current user. Spotify API route is named `/me/profile`

Comment (Iqro-dev) [reply src/modules/spotify/auth/spotify-auth.service.spec.ts:81]: Oh, alright then.

**Review by Iqro-dev — APPROVED** [review]

**Pull request merged**